examples_tf2_py/examples_tf2_py/waits_for_transform.py

```diff
@@ -0,0 +1,61 @@
+#!/usr/bin/env python3
+
+import math
+
+from geometry_msgs.msg import TransformStamped
+import rclpy
+from rclpy.duration import Duration
+from rclpy.node import Node
+from rclpy.time import Time
+from tf2_ros import LookupException
+from tf2_ros.buffer import Buffer
+from tf2_ros.transform_listener import TransformListener
+
+
+class WaitsForTransform(Node):
+    """
+    Wait for a transform syncronously.
+
+    This class is an example of waiting for transforms.
+    This will block the executor if used within a callback.
+    See the async example showing how to use coroutine callbacks to avoid this.
+    """
+
+    def __init__(self):
+        super().__init__('example_waits_for_transform')
+
+        self._tf_buffer = Buffer()
+        self._tf_listener = TransformListener(self._tf_buffer, self)
+
+        self._output_timer = self.create_timer(1.0, self.on_timer)
+
+    def on_timer(self):
+        from_frame = 'sonar_2'
+        to_frame = 'starboard_wheel'
+        self.get_logger().info('Waiting for transform from {} to {}'.format(from_frame, to_frame))
+        try:
+            # Block until the transform is available
+            when = rclpy.time.Time()
+            self._tf_buffer.lookup_transform(
```

Comment:
When *not* using coroutines, what's our recommendations for doing lookups like this?  Is it really to block in the callback?  Obviously this works with a MultiThreadedExecutor, but it seems like blocking up any of the executor threads is problematic.  I'm just wondering whether we should have this example at all, or if we should suggest something else?

Reply:
> When not using coroutines, what's our recommendations for doing lookups like this? Is it really to block in the callback?

I think so. Maybe someone could start a dedicated thread to poll for the transform? It would be nice if there was an api that called a callback when a transform became ready.

> I'm just wondering whether we should have this example at all, or if we should suggest something else?

The API exists and this calls out in docstrings and comments that the blocking behavior is a problem. However, I would always use coroutines instead of the blocking API. I think the example should exist since the comments could steer someone to use the coroutine API instead.

Reply:
> > 
> 
> I think so. Maybe someone could start a dedicated thread to poll for the transform? It would be nice if there was an api that called a callback when a transform became ready.

Oh yeah, good call.

So I'm OK with adding this example, as this is something people will want to do.  But I'll request two things here:
1.  Open an issue to have a callback-based reply to `lookup_transform`.
1.  Update the comment right above this call, or in the class header (your choice) to be a little more strong and suggest to always use the coroutine way instead.

Reply:
It looks like it should be possible to avoid blocking without coroutines, but I encountered an issues ros2/rclpy#460, but that should be fixed by ros2/rclpy#461. I added a second example in 9c74ebc, but it could be made simpler if the result included the transform, so I opened #194 too.